src/navsat_transform.cpp

```diff
@@ -692,7 +706,7 @@ void NavSatTransform::odomCallback(
   world_frame_id_ = msg->header.frame_id;
   base_link_frame_id_ = msg->child_frame_id;
 
-  if (!transform_good_ && !use_manual_datum_) {
+  if (!transform_good_) {
```

Comment:
Removing the check here for use_manual_datum should be fine since setTransformOdometry() also checks this flag.
Removing the check allows setTransformOdometry to store the world and base frame names from the received odometry message.